src/test/java/Homework17.java

```diff
@@ -0,0 +1,139 @@
+//
+//        //Steps
+//        //Step 1
+//        String url = "https://bbb.testpro.io/";
+//        driver.get(url);
+//        //Step 2
+//        WebElement emailField = driver.findElement(By.cssSelector("input[type='email']"));
+//        emailField.click();//not needed
+//        emailField.clear();
+//        emailField.sendKeys("demo@class.com");
+//
+//        WebElement passwordField = driver.findElement(By.cssSelector("input[type='password']"));
+//        passwordField.click();//not needed
+//        passwordField.clear();
+//        passwordField.sendKeys("te$t$tudent");
+//
+//        WebElement submit = driver.findElement(By.cssSelector("button[type='submit']"));
+//        submit.click();
+//
+//        WebElement avatarIcon = driver.findElement(By.cssSelector("img[class='avatar']"));
+//        Assert.assertTrue(avatarIcon.isDisplayed());
+//        line 14 - 36 will be replaced with the ff login code below:
+
+        //lets try first if our login works, then we can proceed to implement the next steps
+        provideEmail("eduardo.santos@testpro.io");
+        providePassword("te$t$tudent");
+        clickSubmit();
+        //our next step would be to search a song, we can try a parameterized approach in searching a song
+        searchSong("epic song");    //now our method to search a song will require a string, let's enter the song we want to search
+        Thread.sleep(2000);//for checking if we are able to search, we can remove this later
+        //looks good, now our next step is to click the View All Button
+        clickViewAllBtn();
+        Thread.sleep(2000);//for checking if we are able to click the view all, we can remove this later
+        //looks good, now our next step is to choose the first song result
+        clickFirstSongResult();
+        Thread.sleep(2000);
+        //lets add a pause to check if the first song is clicked
+        //looks good, the xpath used was correct, but wont work with a different song
+        //for our next step, we will click the add to button, naming readable methods will help us proceed on the steps on our tests
+        //like clickAddToBtn, clickFirstSongResult etc.
+        clickAddToBtn();
+        Thread.sleep(2000);//lets add this to check if add to is clicked
+        //looks good, next will be to choose a playlist to add it to
+        choosePlaylistToAddTo();
+        Thread.sleep(2000);
+        //Last step would be to assert/verify song is added
+
+        String expectedMessage = "Added 1 song into";
+        Assert.assertTrue(getVerificationMessage().contains(expectedMessage));
+        //this assertion verifies if the retrieved text from the notification pop up contains the sequence of characters in the expectedMessage string
+        //it will return true if the words or characters exist
+
+        //for another way to assert/verify, we can check if the notification pop up is present or displayed
+        Assert.assertTrue(isNotificationPopUpDisplayed());
```

Comment:
**Use in test methods, the following logical split:   GIVEN – WHEN - THEN**
    
For example

```
 @Test

    public void testChangeProfileName (){
        // GIVEN

        String newName  = getRandomString();

        String currentPassword = "te$t$tudent";

        provideEmail("demo@class.com");

        providePassword();

        clickSubmitBtn();

 

        // WHEN

        openUserProfilePage();

        setName(newName);

        setCurrentPassword(currentPassword);

 

 

        // THEN

        Assert.assertTrue(getSuccessPopUp().isDisplayed());

}
```
One Test - One Assert